components/tidb_query_datatype/src/codec/datum.rs

```diff
@@ -1085,6 +1085,20 @@ pub fn split_datum(buf: &[u8], desc: bool) -> Result<(&[u8], &[u8])> {
     Ok(buf.split_at(1 + pos))
 }
 
+/// `walk_n_columns` walks `n` colums within `buf` which is in datum format
+/// and advances the buffer pointer.
+/// If the datum buffer contains less than `n` columns, an error will be returned.
+pub fn walk_n_columns(buf: &mut &[u8], n: usize) -> Result<()> {
```

Comment:
`skip_n`? (notice that each datum may not be a column)

Reply:
Done.